catalyst_voices/lib/widgets/rich_text/voices_rich_text.dart

```diff
@@ -0,0 +1,256 @@
+import 'package:catalyst_voices/widgets/buttons/voices_filled_button.dart';
+import 'package:catalyst_voices/widgets/common/resizable_box_parent.dart';
+import 'package:catalyst_voices_assets/catalyst_voices_assets.dart';
+import 'package:catalyst_voices_brands/catalyst_voices_brands.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter_quill/extensions.dart';
+import 'package:flutter_quill/flutter_quill.dart';
+import 'package:flutter_quill_extensions/flutter_quill_extensions.dart';
+
+/// A component for rich text writing
+/// using Quill under the hood
+/// https://pub.dev/packages/flutter_quill
+class VoicesRichText extends StatefulWidget {
+  const VoicesRichText({
+    super.key,
+    this.document,
+    this.onSave,
+    this.charsLimit,
+  });
+
+  final Document? document;
+  final ValueChanged<Document>? onSave;
+  final int? charsLimit;
+
+  @override
+  State<VoicesRichText> createState() => _VoicesRichTextState();
+}
+
+class _VoicesRichTextState extends State<VoicesRichText> {
+  final QuillController _controller = QuillController.basic();
+  int _documentLength = 0;
+  bool _editMode = false;
+  FocusNode _focusNode = FocusNode();
+
+  void _onDocumentChange(DocChange docChange) {
+    final documentLength = _controller.document.length;
+    setState(() {
+      _documentLength = documentLength;
+    });
+    final limit = widget.charsLimit;
+
+    if (limit == null) return;
+
+    if (documentLength > limit) {
+      final latestIndex = limit - 1;
+      _controller.replaceText(
+        latestIndex,
+        documentLength - limit,
+        '',
+        TextSelection.collapsed(offset: latestIndex),
+      );
+    }
+  }
+
+  @override
+  void initState() {
+    super.initState();
+    if (widget.document != null) _controller.document = widget.document!;
+    _controller.document.changes.listen(_onDocumentChange);
+
+    setState(() {
+      _documentLength = _controller.document.length;
+    });
+  }
+
+  @override
+  void dispose() {
+    _controller.dispose();
+    super.dispose();
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    final theme = Theme.of(context);
+
+    return ExcludeFocus(
+      excluding: !_editMode,
+      child: Column(
+        children: [
+          Padding(
+            padding: const EdgeInsets.only(left: 16.0, top: 20, bottom: 20),
```

Comment:
Probably we don't have a lint rule for that but prefer to keep one way of writing numbers, either skip `.0` for all of them or add it everywhere, now it seems a bit inconsistent. I think we can skip it